include-refactoring/xsimd/types/xsimd_utils.hpp

```diff
@@ -323,7 +381,127 @@ namespace xsimd
+    CASE_LHS(EXPAND(op), 6 + addx);      \
+    CASE_LHS(EXPAND(op), 7 + addx);
+
+#define XSIMD_REPEAT_8_N(op, addx)       \
+    CASE_LHS(EXPAND(op), 0 + addx);      \
+    XSIMD_REPEAT_8_0(op, addx);
+
+#define XSIMD_REPEAT_8(op)               \
+    XSIMD_REPEAT_8_0(op, 0);
+
+#define XSIMD_REPEAT_16_0(op, addx)      \
+    XSIMD_REPEAT_8_0(op, 0 + addx);      \
+    XSIMD_REPEAT_8_N(op, 8 + addx);
+
+#define XSIMD_REPEAT_16_N(op, addx)      \
+    XSIMD_REPEAT_8_N(op, 0 + addx);      \
+    XSIMD_REPEAT_8_N(op, 8 + addx);
+
+#define XSIMD_REPEAT_16(op)              \
+    XSIMD_REPEAT_16_0(op, 0);
+
+#define XSIMD_REPEAT_32_0(op, addx)      \
+    XSIMD_REPEAT_16_0(op, 0 + addx);     \
+    XSIMD_REPEAT_16_N(op, 16 + addx);
+
+#define XSIMD_REPEAT_32_N(op, addx)      \
+    XSIMD_REPEAT_16_N(op, 0 + addx);     \
+    XSIMD_REPEAT_16_N(op, 16 + addx);
+
+#define XSIMD_REPEAT_32(op)              \
+    XSIMD_REPEAT_32_0(op, 0);
+
+#define XSIMD_REPEAT_64(op)              \
+    XSIMD_REPEAT_32_0(op, 0);            \
+    XSIMD_REPEAT_32_N(op, 32);
+
+/* The Macro is for vext (lhs, rhs, n)
+ *
+ * _mm_alignr_epi8, _mm_alignr_epi32 ...
+ */
+#define CASE_LHS_RHS(op, i)              \
+    case i: return op(lhs, rhs, i);
+
+#define XSIMD_REPEAT_2_0(op, addx)       \
+    CASE_LHS_RHS(EXPAND(op), 1 + addx);
+
+#define XSIMD_REPEAT_2_N(op, addx)       \
+    CASE_LHS_RHS(EXPAND(op), 0 + addx);  \
+    XSIMD_REPEAT_2_0(op, addx);
+
+#define XSIMD_REPEAT_2(op)               \
+    XSIMD_REPEAT_2_0(op, 0);
+
+#define XSIMD_REPEAT_4_0(op, addx)       \
+    XSIMD_REPEAT_2_0(op, 0 + addx);      \
+    XSIMD_REPEAT_2_N(op, 2 + addx);
+
+#define XSIMD_REPEAT_4_N(op, addx)       \
+    XSIMD_REPEAT_2_N(op, 0 + addx);      \
+    XSIMD_REPEAT_2_N(op, 2 + addx);
+
+#define XSIMD_REPEAT_4(op)               \
+    XSIMD_REPEAT_4_0(op, 0);
+
+#define XSIMD_REPEAT_8_0_v2(op, addx)    \
+    XSIMD_REPEAT_4_0(op, 0 + addx);      \
+    XSIMD_REPEAT_4_N(op, 4 + addx);
+
+#define XSIMD_REPEAT_8_N_v2(op, addx)    \
+    XSIMD_REPEAT_4_N(op, 0 + addx);      \
+    XSIMD_REPEAT_4_N(op, 4 + addx);
+
+#define XSIMD_REPEAT_8_v2(op)            \
+    XSIMD_REPEAT_8_0_v2(op, 0);
+
+#define XSIMD_REPEAT_16_0_v2(op, addx)   \
+    XSIMD_REPEAT_8_0_v2(op, 0 + addx);   \
+    XSIMD_REPEAT_8_N_v2(op, 8 + addx);
+
+#define XSIMD_REPEAT_16_N_v2(op, addx)   \
+    XSIMD_REPEAT_8_N_v2(op, 0 + addx);   \
+    XSIMD_REPEAT_8_N_v2(op, 8 + addx);
+
+#define XSIMD_REPEAT_16_v2(op)           \
+    XSIMD_REPEAT_16_0_v2(op, 0);
+
+#define XSIMD_REPEAT_32_0_v2(op, addx)   \
+    XSIMD_REPEAT_16_0_v2(op, 0 + addx);  \
+    XSIMD_REPEAT_16_N_v2(op, 16 + addx);
+
+#define XSIMD_REPEAT_32_N_v2(op, addx)   \
+    XSIMD_REPEAT_16_N_v2(op, 0 + addx);  \
+    XSIMD_REPEAT_16_N_v2(op, 16 + addx);
+
+#define XSIMD_REPEAT_32_v2(op)           \
+    XSIMD_REPEAT_32_0_v2(op, 0);
 
+#define XSIMD_REPEAT_64_v2(op)           \
+    XSIMD_REPEAT_32_0_v2(op, 0);         \
+    XSIMD_REPEAT_32_N_v2(op, 32);
```

Comment:
The return of the macro... Why????

Reply:
Please use the trick I've been using for Intel arch instead

Reply:
Sorry for that, I've implemented that before you showed me the trick and forgot to make a second pass. I'm gonna fix it.